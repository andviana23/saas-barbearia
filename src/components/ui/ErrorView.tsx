'use client';

import { Box, Typography, Button, Stack, Alert } from '@mui/material';
import { ErrorOutline, Refresh, Home } from '@mui/icons-material';
import { ReactNode } from 'react';

interface ErrorViewProps {
  title?: string;
  message?: string;
  error?: Error | string;
  onRetry?: () => void;
  onHome?: () => void;
  children?: ReactNode;
  variant?: 'page' | 'card' | 'inline';
  severity?: 'error' | 'warning';
}

export default function ErrorView({
  title = 'Ops! Algo deu errado',
  message = 'Ocorreu um erro inesperado. Tente novamente ou entre em contato com o suporte.',
  error,
  onRetry,
  onHome,
  children,
  variant = 'page',
  severity = 'error',
}: ErrorViewProps) {
  const getContainerProps = () => {
    const baseProps = {
      display: 'flex',
      flexDirection: 'column' as const,
      alignItems: 'center',
      textAlign: 'center' as const,
      gap: 3,
    };

    switch (variant) {
      case 'card':
        return {
          ...baseProps,
          p: 4,
          border: '1px solid',
          borderColor: 'divider',
          borderRadius: 2,
          bgcolor: 'background.paper',
        };
      case 'inline':
        return {
          ...baseProps,
          p: 2,
        };
      default: // page
        return {
          ...baseProps,
          minHeight: '60vh',
          justifyContent: 'center',
          p: 4,
        };
    }
  };

  const getErrorDetails = () => {
    if (!error) return null;

    const errorMessage = typeof error === 'string' ? error : error.message;
    const errorStack = typeof error === 'object' ? error.stack : undefined;

    return (
      <Alert severity={severity} sx={{ width: '100%', maxWidth: 600, textAlign: 'left' }}>
        <Typography
          variant="body2"
          component="pre"
          sx={{
            whiteSpace: 'pre-wrap',
            fontFamily: 'monospace',
            fontSize: '0.875rem',
          }}
        >
          {errorMessage}
          {process.env.NODE_ENV === 'development' && errorStack && (
            <>
              {'\n\n'}
              {errorStack}
            </>
          )}
        </Typography>
      </Alert>
    );
  };

  return (
    <Box sx={getContainerProps()}>
      <ErrorOutline
        sx={{
          fontSize: variant === 'inline' ? 48 : 64,
          color: severity === 'error' ? 'error.main' : 'warning.main',
        }}
      />

      <Stack spacing={1} alignItems="center">
        <Typography variant={variant === 'inline' ? 'h6' : 'h4'} component="h1" gutterBottom>
          {title}
        </Typography>

        <Typography variant="body1" color="text.secondary" sx={{ maxWidth: 600 }}>
          {message}
        </Typography>
      </Stack>

      {(onRetry || onHome) && (
        <Stack direction="row" spacing={2} flexWrap="wrap" justifyContent="center">
          {onRetry && (
            <Button variant="contained" startIcon={<Refresh />} onClick={onRetry}>
              Tentar novamente
            </Button>
          )}

          {onHome && (
            <Button variant="outlined" startIcon={<Home />} onClick={onHome}>
              Ir para in√≠cio
            </Button>
          )}
        </Stack>
      )}

      {children}

      {error && getErrorDetails()}
    </Box>
  );
}
