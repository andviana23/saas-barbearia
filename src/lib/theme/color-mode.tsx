'use client';

import { createContext, useContext, useEffect, useMemo, useState, ReactNode } from 'react';
import { ThemeProvider, CssBaseline } from '@mui/material';
import { makeTheme } from './index';

type ColorMode = 'light' | 'dark';

interface ColorModeContextType {
  colorMode: ColorMode;
  toggleColorMode: () => void;
  setColorMode: (mode: ColorMode) => void;
}

const ColorModeContext = createContext<ColorModeContextType | undefined>(undefined);

export function useColorMode() {
  const context = useContext(ColorModeContext);
  if (!context) {
    throw new Error('useColorMode must be used within a ColorModeProvider');
  }
  return context;
}

interface ColorModeProviderProps {
  children: ReactNode;
}

/**
 * Regras:
 * - Default: dark
 * - Lê `localStorage[color-mode]`
 * - Se não houver, usa `prefers-color-scheme`
 * - Aplica `data-theme` no <html> (útil p/ CSS globais)
 * - Aplica CssBaseline e ThemeProvider com makeTheme(mode)
 */
export function ColorModeProvider({ children }: ColorModeProviderProps) {
  const [colorMode, _setColorMode] = useState<ColorMode>('dark');

  // Carrega preferência inicial
  useEffect(() => {
    try {
      const stored = localStorage.getItem('color-mode') as ColorMode | null;
      if (stored === 'light' || stored === 'dark') {
        _setColorMode(stored);
        return;
      }
      const prefersDark = window.matchMedia?.('(prefers-color-scheme: dark)').matches;
      _setColorMode(prefersDark ? 'dark' : 'light');
    } catch {
      // em caso de bloqueio de storage, mantém 'dark'
    }
  }, []);

  // Sincroniza no DOM e no storage
  useEffect(() => {
    // atributo para CSS globais (se precisar)
    document.documentElement.setAttribute('data-theme', colorMode);
    try {
      localStorage.setItem('color-mode', colorMode);
    } catch {
      // ignore
    }
  }, [colorMode]);

  // Reação a mudanças do sistema (se o usuário não tiver escolhido manualmente)
  useEffect(() => {
    const mql = window.matchMedia?.('(prefers-color-scheme: dark)');
    if (!mql) return;

    const handleChange = () => {
      const stored = localStorage.getItem('color-mode');
      // Só segue o sistema se não houver preferência explícita salva
      if (stored !== 'light' && stored !== 'dark') {
        _setColorMode(mql.matches ? 'dark' : 'light');
      }
    };

    mql.addEventListener?.('change', handleChange);
    return () => mql.removeEventListener?.('change', handleChange);
  }, []);

  const theme = useMemo(() => makeTheme(colorMode), [colorMode]);

  const setColorMode = (mode: ColorMode) => {
    _setColorMode(mode);
    try {
      localStorage.setItem('color-mode', mode);
    } catch {
      // ignore
    }
  };

  const toggleColorMode = () => {
    setColorMode(colorMode === 'light' ? 'dark' : 'light');
  };

  return (
    <ColorModeContext.Provider value={{ colorMode, toggleColorMode, setColorMode }}>
      <ThemeProvider theme={theme}>
        <CssBaseline />
        {children}
      </ThemeProvider>
    </ColorModeContext.Provider>
  );
}
