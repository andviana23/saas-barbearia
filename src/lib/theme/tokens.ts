// Radius padronizado (DS v2.1): padrão universal 4px; xs para chips/sutilezas
export const radius = {
  xs: 1,
  sm: 2, // pode ser usado em inputs específicos; próximo mas distinto
  md: 4, // padrão universal (cards, modais, botões)
  lg: 6,
} as const;

export const shadows = {
  none: 'none',
  sm: '0 1px 2px rgba(0,0,0,0.12)',
  md: '0 4px 12px rgba(0,0,0,0.18)',
  lg: '0 8px 24px rgba(0,0,0,0.22)',
} as const;

// Paleta Dark atualizada (Design System)
export const paletteDark = {
  primary: {
    main: '#F4A300',
    light: '#F6B733',
    dark: '#CC8A00',
  },
  secondary: {
    main: '#FFFFFF',
    light: '#FFFFFF',
    dark: '#E5E7EB',
  },
  success: { main: '#22C55E' },
  warning: { main: '#FBBF24' },
  error: { main: '#F87171' },
  info: { main: '#22C55E' },
  divider: 'rgba(255,255,255,0.08)',
  background: {
    default: '#181818',
    paper: '#2D2D2D',
  },
  // @ts-ignore
  surfaces: {
    1: '#2D2D2D',
    2: '#404040',
  },
  text: {
    primary: '#FFFFFF',
    secondary: '#9CA3AF',
    disabled: '#6B7280',
  },
  action: {
    hover: 'rgba(244, 163, 0, 0.08)',
    selected: 'rgba(244, 163, 0, 0.16)',
    disabledOpacity: 0.38,
  },
  neutralBorder: 'rgba(255,255,255,0.1)',
  input: {
    background: '#2D2D2D',
    placeholder: '#6B7280',
  },
} as const;

export const paletteLight = {
  primary: { main: '#F4A300' },
  secondary: { main: '#181818' },
  success: { main: '#16A34A' },
  warning: { main: '#D97706' },
  error: { main: '#DC2626' },
  info: { main: '#2563EB' },
  divider: '#E5E7EB',
  background: {
    default: '#FFFFFF',
    paper: '#FFFFFF',
  },
  text: {
    primary: '#181818',
    secondary: '#6B7280',
    disabled: 'rgba(24,24,24,0.4)',
  },
  action: {
    hover: 'rgba(244, 163, 0, 0.04)',
    selected: 'rgba(244, 163, 0, 0.10)',
    disabledOpacity: 0.38,
  },
  neutralBorder: '#E5E7EB',
} as const;
