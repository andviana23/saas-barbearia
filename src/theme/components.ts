// src/theme/components.ts
import { Components, Theme } from '@mui/material/styles';

export const components = (theme: Theme): Components => ({
  MuiCssBaseline: {
    styleOverrides: {
      body: {
        backgroundColor: theme.palette.background.default, // #080F18 (dark)
        color: theme.palette.text.primary,
        textRendering: 'optimizeLegibility',
        WebkitFontSmoothing: 'antialiased',
        MozOsxFontSmoothing: 'grayscale',
      },
    },
  },

  MuiPaper: {
    defaultProps: { elevation: 0 },
    styleOverrides: {
      root: {
        borderRadius: theme.shape.borderRadius,
        border: `1px solid ${theme.palette.divider}`,
        backgroundColor: theme.palette.background.paper, // #040E18
      },
    },
  },

  MuiCard: {
    styleOverrides: {
      root: {
        borderRadius: theme.shape.borderRadius,
        border: `1px solid ${theme.palette.divider}`,
        backgroundColor: theme.palette.background.paper, // #040E18
      },
    },
  },

  MuiButton: {
    styleOverrides: {
      root: {
        borderRadius: theme.shape.borderRadius,
        textTransform: 'none',
        fontWeight: 600,
        boxShadow: 'none',
      },
    },
  },

  MuiOutlinedInput: {
    styleOverrides: {
      root: {
        borderRadius: theme.shape.borderRadius,
        backgroundColor: theme.palette.mode === 'dark' ? '#05070A' : undefined,
        '& .MuiOutlinedInput-notchedOutline': {
          borderColor: theme.palette.divider,
        },
        '&:hover .MuiOutlinedInput-notchedOutline': {
          borderColor: theme.palette.grey[600],
        },
        '&.Mui-focused .MuiOutlinedInput-notchedOutline': {
          borderColor: theme.palette.primary.main,
        },
        '& input::placeholder': {
          color: theme.palette.grey[400], // #343E51
          opacity: 1,
        },
      },
    },
  },

  MuiListItemButton: {
    styleOverrides: {
      root: {
        borderRadius: theme.shape.borderRadius,
        '&:hover': {
          backgroundColor:
            theme.palette.mode === 'dark' ? theme.palette.grey[500] : theme.palette.action.hover, // #73777E no dark
        },
      },
    },
  },

  MuiMenu: {
    styleOverrides: {
      paper: {
        elevation: 0,
        backgroundColor: theme.palette.background.paper,
        border: `1px solid ${theme.palette.divider}`,
        borderRadius: theme.shape.borderRadius,
      },
    },
  },

  MuiDialog: {
    styleOverrides: {
      paper: {
        borderRadius: theme.shape.borderRadius,
        backgroundColor: theme.palette.background.paper,
        border: `1px solid ${theme.palette.divider}`,
      },
    },
  },
});
