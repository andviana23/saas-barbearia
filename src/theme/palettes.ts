import { PaletteOptions } from '@mui/material/styles';

/**
 * Paleta Light - Design System Atualizado
 */
export const lightPalette: PaletteOptions = {
  mode: 'light',
  primary: {
    main: '#F4A300',
    light: '#F6B733',
    dark: '#CC8A00',
    contrastText: '#FFFFFF',
  },
  secondary: {
    main: '#181818',
    light: '#333333',
    dark: '#000000',
    contrastText: '#FFFFFF',
  },
  background: {
    default: '#FFFFFF',
    paper: '#FFFFFF',
  },
  text: {
    primary: '#181818',
    secondary: '#6B7280',
    disabled: '#E5E7EB',
  },
  divider: '#E5E7EB',
  action: {
    active: '#F4A300',
    hover: 'rgba(244, 163, 0, 0.04)',
    selected: 'rgba(244, 163, 0, 0.12)',
    disabled: '#E5E7EB',
    disabledBackground: '#F9FAFB',
  },
  success: {
    main: '#10B981',
    light: '#34D399',
    dark: '#059669',
    contrastText: '#FFFFFF',
  },
  warning: {
    main: '#F59E0B',
    light: '#FBBF24',
    dark: '#D97706',
    contrastText: '#FFFFFF',
  },
  error: {
    main: '#EF4444',
    light: '#F87171',
    dark: '#DC2626',
    contrastText: '#FFFFFF',
  },
  info: {
    main: '#2563EB',
    light: '#60A5FA',
    dark: '#1D4ED8',
    contrastText: '#FFFFFF',
  },
  grey: {
    50: '#F9FAFB',
    100: '#F3F4F6',
    200: '#E5E7EB',
    300: '#D1D5DB',
    400: '#9CA3AF',
    500: '#6B7280',
    600: '#4B5563',
    700: '#374151',
    800: '#1F2937',
    900: '#111827',
  },
};

/**
 * Paleta Dark - Design System Atualizado
 */
export const darkPalette: PaletteOptions = {
  mode: 'dark',

  primary: {
    main: '#F4A300',
    light: '#F6B733',
    dark: '#CC8A00',
    contrastText: '#181818',
  },

  secondary: {
    main: '#FFFFFF',
    light: '#FFFFFF',
    dark: '#E5E7EB',
    contrastText: '#181818',
  },

  background: {
    default: '#181818',
    paper: '#2D2D2D',
  },

  text: {
    primary: '#FFFFFF',
    secondary: '#9CA3AF',
    disabled: '#6B7280',
  },

  divider: 'rgba(255,255,255,0.08)',

  action: {
    active: '#F4A300',
    hover: 'rgba(244, 163, 0, 0.08)',
    selected: 'rgba(244, 163, 0, 0.16)',
    disabled: '#6B7280',
    disabledBackground: '#374151',
  },

  success: {
    main: '#22C55E',
    light: '#4ADE80',
    dark: '#16A34A',
    contrastText: '#FFFFFF',
  },
  warning: {
    main: '#F59E0B',
    light: '#FBBF24',
    dark: '#D97706',
    contrastText: '#FFFFFF',
  },
  error: {
    main: '#EF4444',
    light: '#F87171',
    dark: '#DC2626',
    contrastText: '#FFFFFF',
  },
  info: {
    main: '#22C55E',
    light: '#4ADE80',
    dark: '#16A34A',
    contrastText: '#FFFFFF',
  },

  grey: {
    50: '#F9FAFB',
    100: '#F3F4F6',
    200: '#E5E7EB',
    300: '#D1D5DB',
    400: '#9CA3AF',
    500: '#6B7280',
    600: '#4B5563',
    700: '#374151',
    800: '#2D2D2D',
    900: '#181818',
  },
};
