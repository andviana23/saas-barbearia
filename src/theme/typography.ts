export const typography = {
  fontFamily: '"Inter", "Roboto", "Helvetica", "Arial", sans-serif',
  h1: {
    fontSize: '2.5rem',
    fontWeight: 700,
    lineHeight: 1.2,
    letterSpacing: '-0.02em',
  },
  h2: {
    fontSize: '2rem',
    fontWeight: 600,
    lineHeight: 1.3,
    letterSpacing: '-0.01em',
  },
  h3: {
    fontSize: '1.75rem',
    fontWeight: 600,
    lineHeight: 1.4,
    letterSpacing: '-0.01em',
  },
  h4: {
    fontSize: '1.5rem',
    fontWeight: 600,
    lineHeight: 1.4,
    letterSpacing: '0em',
  },
  h5: {
    fontSize: '1.25rem',
    fontWeight: 600,
    lineHeight: 1.5,
    letterSpacing: '0em',
  },
  h6: {
    fontSize: '1.125rem',
    fontWeight: 600,
    lineHeight: 1.5,
    letterSpacing: '0em',
  },
  subtitle1: {
    fontSize: '1rem',
    fontWeight: 500,
    lineHeight: 1.5,
    letterSpacing: '0.01em',
  },
  subtitle2: {
    fontSize: '0.875rem',
    fontWeight: 500,
    lineHeight: 1.5,
    letterSpacing: '0.01em',
  },
  body1: {
    fontSize: '1rem',
    fontWeight: 400,
    lineHeight: 1.6,
    letterSpacing: '0.01em',
  },
  body2: {
    fontSize: '0.875rem',
    fontWeight: 400,
    lineHeight: 1.6,
    letterSpacing: '0.01em',
  },
  button: {
    fontSize: '0.875rem',
    fontWeight: 600,
    lineHeight: 1.5,
    letterSpacing: '0.01em',
    textTransform: 'none' as const,
  },
  caption: {
    fontSize: '0.75rem',
    fontWeight: 400,
    lineHeight: 1.5,
    letterSpacing: '0.02em',
  },
  overline: {
    fontSize: '0.75rem',
    fontWeight: 600,
    lineHeight: 1.5,
    letterSpacing: '0.1em',
    textTransform: 'uppercase' as const,
  },
};
