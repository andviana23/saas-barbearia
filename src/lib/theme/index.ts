import { createTheme, ThemeOptions } from '@mui/material/styles';
import { shadows, paletteDark, paletteLight, radius } from './tokens';

const typography: ThemeOptions['typography'] = {
  fontFamily: '"Inter", "Roboto", "Helvetica", sans-serif', // Conforme DS v2
  h1: { fontSize: '2rem', fontWeight: 600 },
  h2: { fontSize: '1.75rem', fontWeight: 600 },
  h3: { fontSize: '1.5rem', fontWeight: 600 },
  h4: { fontSize: '1.25rem', fontWeight: 500 },
  h5: { fontWeight: 600, fontSize: '1.125rem' },
  h6: { fontWeight: 600, fontSize: '1rem' },
  body1: { fontSize: '1rem', fontWeight: 400 },
  body2: { fontSize: '0.875rem', fontWeight: 400 },
  button: { fontWeight: 600, textTransform: 'none' }, // Conforme DS v2
  caption: { fontSize: '0.75rem' },
};

const components = (isDark: boolean, neutralBorder: string): ThemeOptions['components'] => ({
  MuiCssBaseline: {
    styleOverrides: {
      body: {
        textRendering: 'optimizeLegibility',
        WebkitFontSmoothing: 'antialiased',
        MozOsxFontSmoothing: 'grayscale',
        // Fundo global atualizado
        backgroundColor: isDark ? '#0B0E13' : '#F8FAFC',
        // Focus ring acessível (aplicado via data attribute quando necessário)
        '--focus-ring': '#4F8CFF',
      },
    },
  },

  MuiPaper: {
    defaultProps: { elevation: 0 },
    styleOverrides: {
      root: {
        borderRadius: radius.md,
        border: `1px solid ${neutralBorder}`,
        backgroundClip: 'padding-box',
        backgroundColor: isDark ? '#161A23' : undefined, // surface-1
      },
    },
  },

  MuiCard: {
    styleOverrides: {
      root: {
        borderRadius: radius.md,
        border: `1px solid ${neutralBorder}`,
        boxShadow: shadows.sm,
        backgroundColor: isDark ? '#161A23' : undefined, // surface-1
      },
    },
  },

  MuiButton: {
    styleOverrides: {
      root: {
        borderRadius: radius.md,
        paddingInline: 16,
        height: 40,
        fontWeight: 600,
        '&:focus-visible': {
          outline: '2px solid var(--focus-ring)',
          outlineOffset: 2,
        },
      },
      containedPrimary: { boxShadow: shadows.sm },
      outlined: { borderWidth: 1.2 },
    },
  },

  MuiTextField: {
    defaultProps: { size: 'small', fullWidth: true },
  },

  MuiOutlinedInput: {
    styleOverrides: {
      root: {
        borderRadius: radius.md,
        '& fieldset': { borderColor: neutralBorder },
        '&:hover fieldset': { borderColor: isDark ? '#4F8CFF' : '#2563EB' },
        '&.Mui-focused fieldset': { borderWidth: 1.5, borderColor: '#4F8CFF' },
        ...(isDark && { backgroundColor: '#12151D' }),
        '& input::placeholder': {
          color: isDark ? '#3A4353' : 'inherit',
          opacity: 1,
        },
      },
      input: {
        paddingBlock: 10,
        '::placeholder': {
          color: isDark ? '#3A4353' : 'inherit',
          opacity: 1,
        },
      },
    },
  },

  MuiAppBar: {
    styleOverrides: {
      root: {
        borderBottom: `1px solid ${neutralBorder}`,
        boxShadow: 'none',
      },
    },
  },

  MuiDrawer: {
    styleOverrides: {
      paper: {
        borderRight: `1px solid ${neutralBorder}`,
        width: 248,
      },
    },
  },

  // ✅ Hover do sidebar
  MuiListItemButton: {
    styleOverrides: {
      root: {
        borderRadius: radius.md,
        '&:hover': {
          backgroundColor: isDark ? 'rgba(255,255,255,0.05)' : 'rgba(2,6,23,0.04)',
        },
      },
    },
  },

  MuiTableHead: {
    styleOverrides: {
      root: {
        backgroundColor: isDark ? '#1C202B' : undefined, // surface-2
        '& .MuiTableCell-head': {
          fontWeight: 600,
          borderBottom: `1px solid ${neutralBorder}`,
        },
      },
    },
  },

  MuiTableCell: {
    styleOverrides: {
      root: { borderBottom: `1px solid ${neutralBorder}` },
      head: { fontSize: '0.75rem', letterSpacing: 0.5, textTransform: 'uppercase' },
    },
  },

  MuiTableRow: {
    styleOverrides: {
      root: {
        '&:hover': {
          backgroundColor: isDark ? 'rgba(79,140,255,0.08)' : 'rgba(37,99,235,0.06)',
        },
      },
    },
  },

  MuiChip: {
    styleOverrides: {
      root: {
        borderRadius: radius.xs,
        fontWeight: 600,
        fontSize: '0.625rem', // ~10px
        letterSpacing: 0.3,
        '&.MuiChip-filledDefault': {
          backgroundColor: isDark ? '#1C202B' : '#F1F5F9',
        },
      },
    },
  },

  MuiDivider: {
    styleOverrides: { root: { borderColor: neutralBorder } },
  },

  MuiTooltip: {
    styleOverrides: {
      tooltip: {
        backgroundColor: isDark ? '#1C202B' : '#1E293B',
        border: `1px solid ${neutralBorder}`,
        fontSize: '0.75rem',
        padding: '6px 8px',
        borderRadius: radius.sm,
      },
    },
  },

  MuiDialog: {
    styleOverrides: {
      paper: {
        backgroundColor: isDark ? '#1C202B' : undefined, // surface-2
        borderRadius: radius.md,
        border: `1px solid ${neutralBorder}`,
      },
    },
  },
});

export function makeTheme(mode: 'dark' | 'light' = 'dark') {
  const isDark = mode === 'dark';
  const palette = isDark ? paletteDark : paletteLight;

  const neutralBorder: string = palette.divider || 'rgba(255,255,255,0.08)';
  return createTheme({
    palette: { mode, ...palette },
    shape: { borderRadius: radius.md },
    typography,
    components: components(isDark, neutralBorder),
  });
}
