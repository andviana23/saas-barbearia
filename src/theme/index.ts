import { createTheme, ThemeOptions } from '@mui/material/styles';
import { lightPalette, darkPalette } from './palettes';
import { typography } from './typography';
import { components } from './components';
import { shadows } from './shadows';

// Tokens de design system
export const designTokens = {
  spacing: {
    xs: 4,
    sm: 8,
    md: 16,
    lg: 24,
    xl: 32,
    xxl: 48,
  },
  borderRadius: {
    sm: 2,
    md: 4, // Novo padr達o DS v2.0.0
    lg: 8,
    xl: 12,
    xxl: 16,
  },
  shadows: {
    sm: '0 1px 3px rgba(0,0,0,0.12), 0 1px 2px rgba(0,0,0,0.24)',
    md: '0 3px 6px rgba(0,0,0,0.15), 0 2px 4px rgba(0,0,0,0.12)',
    lg: '0 10px 20px rgba(0,0,0,0.15), 0 3px 6px rgba(0,0,0,0.10)',
    xl: '0 15px 25px rgba(0,0,0,0.15), 0 5px 10px rgba(0,0,0,0.05)',
  },
};

// Tema Light
export const lightTheme = createTheme({
  palette: lightPalette,
  typography,
  components: components(createTheme()),
  shadows,
  shape: {
    borderRadius: designTokens.borderRadius.md, // 4px como novo padr達o DS v2.0.0
  },
  spacing: (factor: number) => `${designTokens.spacing.xs * factor}px`,
});

// Tema Dark
export const darkTheme = createTheme({
  palette: darkPalette,
  typography,
  components: components(createTheme()),
  shadows,
  shape: {
    borderRadius: designTokens.borderRadius.md, // 4px como novo padr達o DS v2.0.0
  },
  spacing: (factor: number) => `${designTokens.spacing.xs * factor}px`,
});

// Tema padr達o (dark-first como especificado)
export const theme = darkTheme;

// Tipos para o tema
export type AppTheme = typeof lightTheme;
export type AppThemeOptions = ThemeOptions;
