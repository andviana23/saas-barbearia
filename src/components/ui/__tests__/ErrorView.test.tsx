import { render, screen, fireEvent } from '@testing-library/react';
import { ThemeProvider } from '@mui/material/styles';
import { theme } from '@/theme';
import ErrorView from '../ErrorView';

const renderWithTheme = (component: React.ReactElement) => {
  return render(<ThemeProvider theme={theme}>{component}</ThemeProvider>);
};

describe('ErrorView', () => {
  it('renders with default props', () => {
    renderWithTheme(<ErrorView />);

    expect(screen.getByText('Ops! Algo deu errado')).toBeInTheDocument();
    expect(screen.getByText(/Ocorreu um erro inesperado/)).toBeInTheDocument();
  });

  it('renders with custom title and message', () => {
    const customTitle = 'Erro personalizado';
    const customMessage = 'Mensagem personalizada';

    renderWithTheme(<ErrorView title={customTitle} message={customMessage} />);

    expect(screen.getByText(customTitle)).toBeInTheDocument();
    expect(screen.getByText(customMessage)).toBeInTheDocument();
  });

  it('renders retry button when onRetry is provided', () => {
    const mockRetry = jest.fn();

    renderWithTheme(<ErrorView onRetry={mockRetry} />);

    const retryButton = screen.getByText('Tentar novamente');
    expect(retryButton).toBeInTheDocument();

    fireEvent.click(retryButton);
    expect(mockRetry).toHaveBeenCalledTimes(1);
  });

  it('renders home button when onHome is provided', () => {
    const mockHome = jest.fn();

    renderWithTheme(<ErrorView onHome={mockHome} />);

    const homeButton = screen.getByText('Ir para inÃ­cio');
    expect(homeButton).toBeInTheDocument();

    fireEvent.click(homeButton);
    expect(mockHome).toHaveBeenCalledTimes(1);
  });

  it('renders error details when error is provided', () => {
    const error = new Error('Test error message');

    renderWithTheme(<ErrorView error={error} />);

    expect(screen.getByText('Test error message')).toBeInTheDocument();
  });

  it('renders error details with string error', () => {
    const errorMessage = 'String error message';

    renderWithTheme(<ErrorView error={errorMessage} />);

    expect(screen.getByText(errorMessage)).toBeInTheDocument();
  });

  it('renders inline variant correctly', () => {
    renderWithTheme(<ErrorView variant="inline" />);

    const title = screen.getByText('Ops! Algo deu errado');
    expect(title).toBeInTheDocument();
  });

  it('renders card variant correctly', () => {
    const { container } = renderWithTheme(<ErrorView variant="card" />);

    expect(container.querySelector('.MuiBox-root')).toBeInTheDocument();
  });

  it('renders with warning severity', () => {
    const error = 'Warning message';

    renderWithTheme(<ErrorView error={error} severity="warning" />);

    expect(screen.getByText(error)).toBeInTheDocument();
  });

  it('creates snapshot for page variant', () => {
    const { container } = renderWithTheme(<ErrorView />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
