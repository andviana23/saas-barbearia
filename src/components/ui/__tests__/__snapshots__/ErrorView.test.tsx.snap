// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorView creates snapshot for page variant 1`] = `
<div
  class="MuiBox-root css-57alq7"
>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-wja5b4-MuiSvgIcon-root"
    data-testid="ErrorOutlineIcon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2M12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8"
    />
  </svg>
  <div
    class="MuiStack-root css-6rg9iv-MuiStack-root"
  >
    <h1
      class="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom css-4p0ogv-MuiTypography-root"
    >
      Ops! Algo deu errado
    </h1>
    <p
      class="MuiTypography-root MuiTypography-body1 css-gb4bb9-MuiTypography-root"
    >
      Ocorreu um erro inesperado. Tente novamente ou entre em contato com o suporte.
    </p>
  </div>
</div>
`;
